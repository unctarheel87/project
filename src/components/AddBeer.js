import React, { Component } from 'react';
import API from '../utils/API';
import { add_Beer, updateMsg } from '../actions';
import { connect } from 'react-redux';

export class AddBeer extends Component {
  state = {
    brewery_name: '',
    beer_name: '',
    beer_style: '',
    ABV: '',
    IBU: ''
  }
  handleChange = name => event => {
    this.setState({
      [name]: event.target.value,
    });
  };
  handleSubmit = event => {
    event.preventDefault();
    const newBeer = {
      "Brewery Name": this.state.brewery_name,
      "Beer Name": this.state.beer_name,
      "Beer Style": this.state.beer_style,
      ABV: this.state.ABV,
      IBU: this.state.IBU
    }
    API.addBeer(newBeer)
      .then(response => {
        this.props.addBeer(response.data)
        this.props.updateMsg('New beer successfully added!')
      }).catch(err => {
        console.log(err);
      })
  }
  render() {
    return (
      <div>
        <form className="uk-flex" onSubmit={this.handleSubmit}>
          <input className="uk-input" type="text" placeholder="Brewery Name" value={this.state.brewery_name} onChange={this.handleChange('brewery_name')} />
          <input className="uk-input" type="text" placeholder="Beer Name"  value={this.state.beer_name} onChange={this.handleChange('beer_name')} />
          <input className="uk-input" type="text" placeholder="Beer Style" value={this.state.beer_style} onChange={this.handleChange('beer_style')} />
          <input className="uk-input" type="text" placeholder="ABV" value={this.state.ABV} onChange={this.handleChange('ABV')} />
          <input className="uk-input" type="text" placeholder="IBU" value={this.state.IBU} onChange={this.handleChange('IBU')} />
          <button className="uk-button uk-button-primary" type="submit">Add</button>
        </form>
        <p>{this.props.msg}</p>
      </div>
    )   
  }
}

const mapStateToProps = state => {
  return {
    msg: state.msg
  }
}

const mapDispatchToProps = dispatch => {
  return {
    addBeer: beer => dispatch(add_Beer(beer)),
    updateMsg: msg => dispatch(updateMsg(msg))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(AddBeer);