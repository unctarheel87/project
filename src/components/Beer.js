import React from 'react';
import API from '../utils/API';
import { connect } from 'react-redux';
import { remove_Beer, update_Beer } from '../actions';

const updateBeer = (id, currentDraft, dispatch) => () => {
  currentDraft = currentDraft ? false : true;
  API.updateBeer(id, { currentDraft })
    .then(response => {
      console.log(response);
      dispatch(update_Beer(id, currentDraft));
    }).catch(err => {
      console.log(err);
    })
}

const removeBeer = (id, dispatch) => () => {
  API.removeBeer(id)
    .then(response => {
      console.log(response);
      dispatch(remove_Beer(id));
    }).catch(err => {
      console.log(err);
    })
}

const green = {
  background: 'lightgreen'
}

const Beer = ({
  dispatch, 
  "Brewery Name": brewery_name, 
  "Beer Name": beer_name, 
  "Beer Style": beer_style, 
  ABV, 
  IBU,
  currentDraft,
  _id, 
}) => (
    <tr>
      <td>{brewery_name}</td>
      <td>{beer_name}</td>
      <td>{beer_style}</td>
      <td>{ABV}</td>
      <td>{IBU}</td>
      <td><button onClick={removeBeer(_id, dispatch)}>X</button></td>
      <td><button 
        style={currentDraft ? green : null}
        onClick={updateBeer(_id, currentDraft, dispatch)}
        >*</button></td>
    </tr>
);

export default connect()(Beer);
