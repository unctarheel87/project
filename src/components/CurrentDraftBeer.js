import React from 'react';

export default ({
  "Brewery Name": brewery_name, 
  "Beer Name": beer_name, 
  "Beer Style": beer_style, 
  ABV, 
  IBU
}) => (
    <tr>
      <td>{brewery_name}</td>
      <td>{beer_name}</td>
      <td>{beer_style}</td>
      <td>{ABV}</td>
      <td>{IBU}</td>
    </tr>
);