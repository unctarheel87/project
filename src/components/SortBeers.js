import React from 'react';
import 
  { sortByBreweryName, 
    sortByBeerName, 
    sortByABV, 
    sortByDraft,
    setTextFilter,
   } from '../actions';
import { connect } from 'react-redux';

const handleChange = props => e => {
  if (e.target.value === "brewery_name") {
    props.sortByBreweryName();
  } else if (e.target.value === "beer_name") {
    props.sortByBeerName();
  } else if (e.target.value === "abv") {
    props.sortByABV();
  } else if (e.target.value === "draft") {
    props.sortByDraft();
  }
};

const onTextChange = props => e => {
  props.setTextFilter(e.target.value)
};

export const SortBeers = props => (
  <form className="uk-width-1-4 uk-margin uk-flex">
    <input
      type="text"
      className="uk-input"
      value={props.filters.text}
      onChange={ onTextChange(props) }
      placeholder="Search"
    />
    <select className="uk-select" value={props.filters.sortBy} onChange={ handleChange(props) }>
      <option value="">filter</option>
      <option value="brewery_name">Brewery Name</option>
      <option value="beer_name">Beer Name</option>
      <option value="abv">ABV</option>
      <option value="draft">Current Draft</option>
    </select>
  </form>
)   

const mapStateToProps = state => {
  return {
    filters: state.filters
  }
}

const mapDispatchToProps = dispatch => {
  return {
    sortByBreweryName: () => dispatch(sortByBreweryName()),
    sortByBeerName: () => dispatch(sortByBeerName()),
    sortByABV: () => dispatch(sortByABV()),
    sortByDraft: () => dispatch(sortByDraft()),
    setTextFilter: (text) => dispatch(setTextFilter(text))
  }
}

export default connect(mapStateToProps, mapDispatchToProps)(SortBeers);