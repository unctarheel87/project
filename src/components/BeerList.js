import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import CurrentDraftBeer from './currentDraftBeer';

const size = {
  position: "relative",
  width: "80%",
  padding: "15px",
  margin: "auto",
  background: "rgba(255, 255, 255, 0.9",
  top: "20vh",
  borderRadius: "5px"
}

const background = {
  background: "url('/images/beertap.jpeg')",
  backgroundSize: "cover",
  height: "100vh"
}

const linkStyle = {
  position: "absolute",
  right: "15px",
  top: "10px",
  color: "whitesmoke"
}

const BeerList = (props) => (
  <section style={ background }>
    <Link className="uk-flex uk-around" style={ linkStyle } to="/browse">
      <i className="material-icons uk-margin-small-right">table_chart</i>
      Admin Panel
    </Link>
    <div style={ size }>
      <h1 className="uk-text-center">Draft List</h1>
      <table className="uk-table uk-table-small uk-table-divider"> 
        <thead>
          <tr>
            <th>Brewery Name</th>
            <th>Beer Name</th>
            <th>Beer Style</th>
            <th>ABV</th>
            <th>IBU</th>
          </tr>
        </thead>
        <tbody>
          { props.beers.length > 0 ?
            props.beers.map(beer => (
              <CurrentDraftBeer key={beer._id} {...beer}/>
            )) :
            <tr className="uk-text-center uk-margin-large-top"><td>Loading...</td></tr>
          }
        </tbody>
      </table>
    </div>
  </section>
)

const mapStateToProps = state => {
  return {
    beers: state.beers.filter(beer => beer.currentDraft)
  }
}

const mapBeerList = connect(mapStateToProps)(BeerList);

export default mapBeerList;
