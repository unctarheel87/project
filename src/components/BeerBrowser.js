import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import Beer from './Beer';
import AddBeer from './AddBeer';
import SortBeers from './SortBeers';
import selectBeers from '../selectors/sortBeers';

const size = {
  width: "90%",
  margin: "15px auto"
}

const linkStyle = {
  position: "absolute",
  right: "15px",
  top: "10px",
  color: "black"
}

const BeerBrowser = props => (
  <div style={ size }>
    <Link className="uk-flex uk-around" style={ linkStyle } to="/">
      <i className="material-icons uk-margin-small-right">format_list_bulleted</i>
      Draft List
    </Link>
    <h1 className="uk-text-center uk-margin-medium">Beer Database</h1>
    <AddBeer />
    <SortBeers />
    <table className="uk-table uk-table-divider">
      <thead>
        <tr>
          <th>Brewery Name</th>
          <th>Beer Name</th>
          <th>Beer Style</th>
          <th>ABV</th>
          <th>IBU</th>
        </tr>
      </thead>
      <tbody>
        { props.beers.length > 0 ?
          props.beers.map(beer => (
            <Beer key={beer._id} {...beer}/>
          )) :
          <tr className="uk-text-center uk-margin-large-top"><td>Loading...</td></tr>
        }
      </tbody>
    </table>
  </div>
);

const mapStateToProps = state => {
  return {
    beers: selectBeers(state.beers, state.filters)
  }
}

const mapBeerBrowser = connect(mapStateToProps)(BeerBrowser);

export default mapBeerBrowser;