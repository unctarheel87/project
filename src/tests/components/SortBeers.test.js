import React from 'react';
import { shallow } from 'enzyme';
import { SortBeers } from '../../components/SortBeers';

const filters = {
  text: '',
  sortBy: '',
};

const fakeFilters = {
  text: 'duck',
  sortBy: 'beer_name',
};

let setTextFilter, sortByBeerName, wrapper

beforeEach(() => {
  setTextFilter = jest.fn();
  sortByBeerName = jest.fn();
  wrapper = shallow(
    <SortBeers
      filters={filters}
      setTextFilter={setTextFilter}
      sortByBeerName={sortByBeerName}
    />
  );
});

test('should render SortBeers correctly', () => {
  expect(wrapper).toMatchSnapshot();
});

test('should render with fake props', () => {
  wrapper.setProps({
    filters: fakeFilters
  });
  expect(wrapper).toMatchSnapshot();
});

test('should handle text change', () => {
  const value = 'duck';
  wrapper.find('input').simulate('change', {
    target: { value }
  });
  expect(setTextFilter).toHaveBeenLastCalledWith(value);
});

test('should sort by "beer_name"', () => {
  const value = 'beer_name';
  wrapper.setProps({
    filters: fakeFilters
  });
  wrapper.find('select').simulate('change', {
    target: { value }
  });
  expect(sortByBeerName).toHaveBeenCalled();
});