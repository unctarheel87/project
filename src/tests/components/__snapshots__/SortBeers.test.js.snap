// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SortBeers correctly 1`] = `
<form
  className="uk-width-1-4 uk-margin uk-flex"
>
  <input
    className="uk-input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value=""
  />
  <select
    className="uk-select"
    onChange={[Function]}
    value=""
  >
    <option
      value=""
    >
      filter
    </option>
    <option
      value="brewery_name"
    >
      Brewery Name
    </option>
    <option
      value="beer_name"
    >
      Beer Name
    </option>
    <option
      value="abv"
    >
      ABV
    </option>
    <option
      value="draft"
    >
      Current Draft
    </option>
  </select>
</form>
`;

exports[`should render with fake props 1`] = `
<form
  className="uk-width-1-4 uk-margin uk-flex"
>
  <input
    className="uk-input"
    onChange={[Function]}
    placeholder="Search"
    type="text"
    value="duck"
  />
  <select
    className="uk-select"
    onChange={[Function]}
    value="beer_name"
  >
    <option
      value=""
    >
      filter
    </option>
    <option
      value="brewery_name"
    >
      Brewery Name
    </option>
    <option
      value="beer_name"
    >
      Beer Name
    </option>
    <option
      value="abv"
    >
      ABV
    </option>
    <option
      value="draft"
    >
      Current Draft
    </option>
  </select>
</form>
`;
